planner/core/testdata/integration_serial_suite_out.json

```diff
@@ -1776,6 +1776,43 @@
           "        └─Selection 9990.00 batchCop[tiflash]  not(isnull(test.t.id))",
           "          └─TableFullScan 10000.00 batchCop[tiflash] table:t keep order:false, stats:pseudo"
         ]
+      },
+      {
+        "SQL": "desc format = 'brief' select /*+hash_agg()*/ sum(id) from (select value, id from t where id > value group by id, value)A group by value /*the exchange should have only one partition column: test.t.value*/",
+        "Plan": [
+          "TableReader 6400.00 root  data:ExchangeSender",
+          "└─ExchangeSender 6400.00 batchCop[tiflash]  ExchangeType: PassThrough",
+          "  └─Projection 6400.00 batchCop[tiflash]  Column#4",
+          "    └─HashAgg 6400.00 batchCop[tiflash]  group by:test.t.value, funcs:sum(test.t.id)->Column#4",
+          "      └─Projection 6400.00 batchCop[tiflash]  test.t.id, test.t.value",
+          "        └─HashAgg 6400.00 batchCop[tiflash]  group by:test.t.id, test.t.value, funcs:firstrow(test.t.id)->test.t.id, funcs:firstrow(test.t.value)->test.t.value",
```

Comment:
The lower Agg doesn't choose a two phase agg, so this doesn't really cover your case.

Reply:
But the next test covers it, then that's ok.